packages/qwik/src/core/use/use-resource.ts

```diff
@@ -252,10 +252,15 @@ export interface ResourceProps<T> {
  */
 // </docs>
 export const Resource = <T>(props: ResourceProps<T>): JSXOutput => {
-  const isBrowser = !isServerPlatform();
+  // Resource path
+  return jsx(Fragment, {
+    children: getResourceValueAsPromise(props),
+  });
+};
+function getResourceValueAsPromise<T>(props: ResourceProps<T>): Promise<JSXOutput> | JSXOutput {
   const resource = props.value as ResourceReturnInternal<T> | Promise<T> | Signal<T>;
-  let promise: Promise<T> | undefined;
-  if (isResourceReturn(resource)) {
+  if (isResourceReturn(resource) && resource.value) {
```

Comment:
Do we need to add the `&& resource.value`?

Reply:
> Do we need to add the `&& resource.value`?

Maybe resource is signal @maiieul

Reply:
the resource is always a promise, no?